### ISSUE TITLE
Problem with `parse` method when `zod.transform` is used in a schema

### ISSUE THREAD

**Issue by piotar** [sam-goodwin/zod-class#31]
The problem occurs when `transform` has been used in the schema.

I don't know what the intention was, but when parsing the schema, `zod class` performs double parsing. In the `parse` function(https://github.com/sam-goodwin/zod-class/blob/main/src/index.ts#L305) and in the constructor(https://github.com/sam-goodwin/zod-class/blob/main/src/index.ts#L187).

I think this is also an performance issue, because why would we parse the schema twice?

There is also a problem of types in constructor when we use `transform` from `zod`.

Example code to reproduce problem:

https://stackblitz.com/edit/stackblitz-starters-cg4bsqz2?file=main.ts&view=editor

```typescript
import { z } from 'zod';
import { Z } from 'zod-class';

enum Color {
  Red,
  Blue,
  Green,
}

enum ColorAsString {
  Red = 'red',
  Blue = 'blue',
  Green = 'green',
}

const ColorMap = {
  [ColorAsString.Red]: Color.Red,
  [ColorAsString.Blue]: Color.Blue,
  [ColorAsString.Green]: Color.Green,
} satisfies Record<ColorAsString, Color>;

class CustomSchema extends Z.class({
  color: z.nativeEnum(ColorAsString).transform((c) => ColorMap[c]),
}) {}

// Works fine, but types are incorrect
console.log(new CustomSchema({ color: ColorAsString.Blue }));

// Throws an exception because it parses twice
console.log(CustomSchema.parse({ color: ColorAsString.Blue }));
```

**Comment by netroy** [comment]
Attempted to fix it [here](https://github.com/sam-goodwin/zod-class/pull/32).

**Comment by sam-goodwin** [comment]
Thanks @netroy! Now available in `0.0.16`
